pymaster/field.py

```diff
@@ -658,3 +659,240 @@ def get_ell_sampling(self):
+        self.ainfo = ut.NmtAlmInfo(lmax)
+        self.field = np.array(field, dtype=np.float64)
+        # Sanity checks
+        if spin is None:
+            spin = 0 if field.ndim == 0 else 2
+        if spin == 2 and np.shape(self.field) != (2, len(self.weights)):
+            raise ValueError("Field has wrong shape.")
+        if spin == 0 and (self.field.ndim != 1
+                          or len(self.field) != len(self.weights)):
+            raise ValueError("Field has wrong shape.")
+        self.spin = spin
+        self.nalms = 2 if spin else 1
+        if not field_is_weighted:
+            self.field *= self.weights
+        if lmax <= 0:
+            lmax = hp.Alm.getlmax(np.shape(self.alm)[-1])
+        self.alm = ut._catalog2alm_ducc0(self.field, self.positions,
+                                         spin=spin, lmax=lmax)
+
+        # 3. Compute Poisson and shear noise bias on mask pseudo-C_ell
+        self.Nw = np.sum(self.weights**2.)/(4.*np.pi)
+        if spin == 2:
+            self.Ngamma = np.sum(
+                (self.field[0]**2. + self.field[1]**2)/(8.*np.pi)
+            )
+
+    def is_compatible(self, other):
+        """ Returns ``True`` if the  of this :obj:`NmtField`
+        is compatible with that of another one (``other``).
+        """
+        if self.minfo != other.minfo:
+            return False
+        if self.ainfo != other.ainfo:
+            return False
+        if self.ainfo_mask != other.ainfo_mask:
+            return False
+        return True
+
+    def get_mask(self):
+        """ Get this field's mask.
+
+        Returns:
+            (`array`): 1D array containing the field's mask.
+        """
+        return self.mask
+
+    def get_maps(self):
+        """ Get this field's set of maps. The returned maps will be
+        masked, contaminant-deprojected, and purified (if so required
+        when generating this :obj:`NmtField`).
+
+        Returns:
+            (`array`): 2D array containing the field's maps.
+        """
+        if self.maps is None:
+            raise ValueError("Input maps unavailable for this field")
+        return self.maps
+
+    def get_alms(self):
+        """ Get the :math:`a_{\\ell m}` coefficients of this field. They
+        include the effects of masking, as well as contaminant deprojection
+        and purification (if required when generating this
+        :obj:`NmtField`).
+
+        Returns:
+            (`array`): 2D array containing the field's :math:`a_{\\ell m}` s.
+        """
+        if self.alm is None:
+            raise ValueError("Mask-only fields have no alms")
+        return self.alm
+
+    def get_mask_alms(self):
+        """ Get the :math:`a_{\\ell m}` coefficients of this field's mask.
+
+        Returns:
+            (`array`): 1D array containing the mask :math:`a_{\\ell m}` s.
+        """
+        return self.alm_mask
+
+    def get_shear_noise_bias(self):
+        """
+        """
+        return self.Ngamma
```

Comment:
Make it more general than `shear`. Maybe just `get_discrete_noise` or something like that.